node/external/transactionAPI/refundDetector.go

```diff
@@ -0,0 +1,45 @@
+package transactionAPI
+
+import (
+	"bytes"
+	"encoding/hex"
+
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+type refundDetectorInput struct {
+	Value         string
+	Data          []byte
+	ReturnMessage string
+	GasLimit      uint64
+}
+
+type refundDetector struct {
+}
+
+func newRefundDetector() *refundDetector {
+	return &refundDetector{}
+}
+
+// Also see: https://github.com/ElrondNetwork/elastic-indexer-go/blob/master/process/transactions/scrsDataToTransactions.go
+func (detector *refundDetector) isRefund(result refundDetectorInput) bool {
+	hasValue := result.Value != "0" && result.Value != ""
+	hasNoGasLimit := result.GasLimit == 0
+	hasReturnCodeOK := detector.isReturnCodeOK(result.Data)
+	isRefundForRelayTxSender := result.ReturnMessage == gasRefundForRelayerMessage
```

Comment:
maybe `strings.Contains` ? If in the future another thing is added to the message, the code will still work

Reply:
Fixed.